test/Microsoft.AspNetCore.SignalR.Redis.Tests/RedisProtocolTests.cs

```diff
@@ -84,22 +84,24 @@ public void WriteGroupCommand(string testName)
         }
 
         // The actual invocation message doesn't matter
-        private static InvocationMessage _testMessage = new InvocationMessage("target", null, Array.Empty<object>());
-        private static Dictionary<string, ProtocolTestData<RedisInvocation>> _invocationTestData = new[]
+        private static InvocationMessage _testMessage = new InvocationMessage("target", Array.Empty<object>());
+
+        // We use a func so we are guaranteed to get a new SerializedHubMessage for each test
```

Comment:
I noticed some subtle issues here when re-running tests. The SerializedHubMessage was being reused between tests. I'll do some investigation, but this Func change fixes the issue.

Reply:
I don't feel strongly about it but instead of making that one piece a func I'd rather the invocation test data was completed recreated each time, i.e. a new copy is returned from a method when required rather than stored as a field.

Reply:
I'm OK with that approach, it's a little cleaner.